starknet_py/proxy/contract_abi_resolver.py

```diff
@@ -97,34 +102,58 @@ async def resolve_abi(self) -> Abi:
         :raises ProxyResolutionError: when given ProxyChecks were not sufficient to resolve proxy
         :raises AbiNotFoundError: when abi is not present in proxied contract class at address
         """
-        impl = await self._get_implementation_from_proxy()
-        implementation, implementation_type = impl
-
-        if implementation_type == ImplementationType.CLASS_HASH:
-            contract_class = await self.client.get_class_by_hash(implementation)
-        else:
-            contract_class = await self._get_class_by_address(address=implementation)
+        implementation_generator = self._get_implementation_from_proxy()
+        async for implementation, implementation_type in implementation_generator:
```

Comment:
Does someone have a better idea for the name of the `implementation` variable? I do not have one, but `implementation = address | class_hash` and it wasn't obvious for me :/ maybe a comment will be enough